pkg/agent/openflow/client.go

```diff
@@ -236,3 +252,57 @@ func (c *client) Initialize(roundNum uint64) error {
 	}
 	return nil
 }
+
+func (c *client) Initialize(roundNum uint64) (<-chan struct{}, error) {
+	// Initiate connections to target OFswitch, and create tables on the switch.
+	connCh := make(chan struct{})
+	if err := c.bridge.Connect(maxRetryForOFSwitch, connCh); err != nil {
+		return nil, err
+	}
+
+	<-connCh
+
+	c.cookieAllocator = cookie.NewAllocator(roundNum)
+
+	return connCh, c.initialize()
+}
+
+func (c *client) Reconcile(roundNum uint64) {
+	c.reconcileMutex.Lock()
+	defer c.reconcileMutex.Unlock()
+
+	c.cookieAllocator = cookie.NewAllocator(roundNum)
+
+	if err := c.initialize(); err != nil {
+		klog.Errorf("Error during flow replay: %v", err)
+	}
+
+	addFixedFlows := func(flows []binding.Flow) {
+		for _, flow := range flows {
+			flow.Reset()
+			if err := c.flowOperations.Add(flow); err != nil {
```

Comment:
Since c.cookieAllocator has been refreshed, would `c.flowOperations.Add` use the cached cookie or the new one when replaying the flows? If cached cookie, would it lead to inconsistent cookieID when new flows are installed?

Reply:
Thanks for catching this, you are correct. I have decided not to change the round number and keep using the same cookie allocator (see latest commit). The rationale is that increasing the round number for a reconnection (as opposed to an agent restart) should not be required, as all the flows have been wiped out from OVS and there is no possibility of stale flows. It makes the code easier as well since I no longer need to query OVSDB for the previous round number. Does that sound reasonable to you or do you think the round number needs to be increased for this case?

Reply:
Do we know ovs-vswitch restarting is the only reason of disconnection? If yes, then I agree keeping same cookie is fine. If not, is there a risk that some flows that should be deleted would be left there as we don't replay deleting flows?
I also think of the case of antrea-agent restarting alone, should we delete the previous flows too, or it has been done?

Reply:
> Do we know ovs-vswitch restarting is the only reason of disconnection?

I cannot think of a situation at the moment where we would have a disconnection but previous flows would be preserved. If I'm wrong, it should be easy to update the cookie in the future.

> I also think of the case of antrea-agent restarting alone, should we delete the previous flows too, or it has been done?

I discussed this with Jianjun yesterday. We are not doing this at the moment, but we should. We can open an issue to track this. @wenyingd do we have a convenient way at the moment to delete all the flows for which the cookie does not match the current round number?

Reply:
If we need two implementations, should we call your current funcs to something like resyncFlows, instead of reconcileFlows? They are not really reconciliation (maybe I did not understand reconciliation correctly though)?

Reply:
reconciliation is maybe a bit misleading since we never read the actual state (no point since it's empty). It's still appropriate IMO since the desire is to reach consistency between the OVS flows and the Antrea flow cache, but I don't mind renaming the method to "replayFlows". I wouldn't say we have "two implementations" though.

Reply:
replayFlows sounds better to me. I just meant later we might have other code for reconciliation at Agent restart, and here at least one difference is that we do not remove stale flows.

Reply:
I'll change the name. For Agent restart, the only thing we need to do is delete stale flows, which should be a couple of lines of code. Flow replay is handled by controllers on agent restart in a very natural way.

Reply:
@antoninbas I have exposed an API `DeleteFlowsByCookie`  to support deleting all Openflow entries using the CookieID/CookieMask. You can call it to remove the stale flows with the Cookie recoreded in the last round (https://github.com/vmware-tanzu/antrea/blob/master/pkg/ovs/openflow/interfaces.go#L68).

Reply:
Sounds good to me.